README.md

```diff
@@ -2,50 +2,89 @@
 [Cryptoanarchy Debian Repository](https://github.com/debian-cryptoanarchy/cryptoanarchy-deb-repo-builder) may help.
-If you install `bitcoin-timechain-regtest` you don't need any further configuration.
+
 Otherwise configure the wallet in `~/.config/pico-bitcoin-wallet/config.toml`.
 
-Create a wallet in bitcoind and mine some coins (mine at least 101 blocks so that coins are mature).
+Example minimal `bitcoin.conf`
+```
+chain=regtest
+# Password for JSON-RPC connections
+rpcpassword=pass
+# Username for JSON-RPC connections
+rpcuser=admin
+```
+
+### Configure pico-bitcoin-wallet
 
-You should implement these commands:
+If you used `bitcoin-timechain-regtest` from `cryptoanarchy` you don't need any further
+configuration.
 
-* `address` - returns the address of the wallet - create it from private key.
-* `scan` - scans the blocks to find matching transactions, should also tell the user how many coins were found.
-* `send` - creates a transaction that sends a given amount to the address provided in the input, bonus: implement BIP21
+Otherwise create `~/.config/pico-bitcoin-wallet/config.toml` (exact location will vary depending on your OS).
 
-First version can be really very lazy if you like. E.g. require scan after sending transaction.
+```toml
+bitcoind_uri = "http://localhost:18443"
+bitcoind_username = "admin"
+bitcoind_password = "pass"
+```
+
+### Mine coins
+
+After you have completed step 3 (implementing the `address` command) you can use `bitcoin-cli
+generatetoaddress 101 <address>` to mine some coins. This is the only `bitcoin-cli` command you need
+to use to complete this tutorial - all other interaction with `bitcoind` is done via code using RPC.
 
 ### Tips and possible gotchas
 
-Things that were discovered during implementation of solution that seem important to mention:
+Things that were discovered during implementation of a solution that seem important to mention:
 
 * Note that the last block has number equal to block count, so the traditional `last_height..current_height` misses the last one.
-* Private key pair for Taproot needs to be tweked before signing - call `.tap_tweak()` on it and then convert it using `to_inner`.
+* Private key pair for Taproot needs to be tweaked before signing - call `.tap_tweak()` on it and then convert it using `to_inner`.
 * To obtain a block from bitcoind you need to first get its hash using `get_block_hash`.
-* `store_txos` takes an iterator, don't bother with being clever if you are low on time - just collect to `Vec` and pass it in.
-* Don't bother with coin selection - just always spend all inputs, that'd be the best anyway if you actually reuse addresses.
+* `store_txos` takes an iterator, don't bother being clever if you are low on time - just collect to `Vec` and pass it in.
+* Don't bother with coin selection, you can either just always spend all inputs or just spend the
+  first one you find that is big enough (they'll be 50BTC each remember).
```

Comment:
Only coinbase inputs will be. So picking first is confusing. Also this drops a comment about spending all inputs with same address being the best approach, but thinking about it it's not always the best and the nuance is too overwhelming.

Reply:
Agree on the coinbase stuff (as mentioned above). Agree also that mentioning "best approach" is a can of worms best left closed.